### PR TITLE
feature: Local OpenQASM Simulator example notebook

### PR DESCRIPTION
**Pull request by ajberdy**
Issue #, if available:

Description of changes:

Update OpenQASM example notebook to reference LocalSimulator.
Create example notebook for OpenQASM LocalSimulator
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+    "\n",
+    "device = LocalSimulator()\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
```
Comment (virajvchaudhari) [inline examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:51]: I think it shouldn't show anything related to PyCharm in the metadata, can you confirm if this aligns with out metadata structure for other notebooks.

File: examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython2",
+   "version": "2.7.6"
+  }
+ },
+ "nbformat": 4,
```
Comment (virajvchaudhari) [inline examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:772]: Similarly, this should have conda_braket as the kernel as we have for other notebooks.

File: examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb (hunk 1 above)
Comment (ajberdy) [reply examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb]: Fixed by importing notebook into an NBI

File: examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb (hunk 2 above)
Comment (ajberdy) [reply examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb]: Fixed by exporting notebook from NBI

File: examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb [hunk 3]
```diff
@@ -138,15 +138,16 @@
    "source": [
```
Comment (virajvchaudhari) [inline examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb:138]: <p>Are we using SV1 device for this now? If not we can replace the text with whichever we intend to use. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (virajvchaudhari) [inline examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb:138]: <p>We are using LocalSimulator() device to run the tasks, but we have instances of SV1 device properties over all notebook cells. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (virajvchaudhari) [inline examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb:138]: <p>Why do we use Aspen-M-1 device here? Should we be replacing that as well? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (virajvchaudhari) [inline examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb:138]: <p>I feel this title is redundant and unnecessary "OpenQASM features on Braket"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (ajberdy) [reply examples/braket_features/Getting_Started_with_OpenQASM_on_Braket.ipynb:138]: <p>This is an already existing notebook. I only changed one cell to show that you can use the LocalSimulator. The rest of the notebook uses SV1 still.</p>

**Comment by virajvchaudhari** [comment]
I think we should also add the open_qasm notebooks in the sub-folder since we have other files along with it i.e. adder.qasm.

**Review by virajvchaudhari — APPROVED** [review]

**Review by virajvchaudhari — APPROVED** [review]

**Review by christianbmadsen — APPROVED** [review]

File: examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb [hunk 4]
```diff
@@ -0,0 +1,733 @@
+{
```
Comment (speller26) [inline examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    pow(p1) @ pow(p2) @ x q;    // sqrt x
</pre><p>I don't think this is really <code>sqrt x</code> :P</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (speller26) [inline examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    gate majority a, b, c {
</pre><p>Should we actually invoke this gate too?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/176/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (ajberdy) [reply examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:1]: <p>Ah right, I'll change that comment</p>

Comment (ajberdy) [reply examples/braket_features/Simulating_Advanced_OpenQASM_Programs_with_the_Local_Simulator.ipynb:1]: <p>Yeah I'll make that change</p>

**Comment by speller26** [comment]
Do we want to explain that the `measure` instruction is redundant and optional (or is that already covered in the other OpenQASM notebook)?

**Comment by ajberdy** [comment]
> Do we want to explain that the `measure` instruction is redundant and optional (or is that already covered in the other OpenQASM notebook)?

I think the examples demonstrate you can get the measurement counts and/or result types without specifying any measurements, also the device properties show that all qubits are always measured. If you think it would be helpful, I could add something

**Review by speller26 — APPROVED** [review]

**Pull request merged**